instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/action_controller_subscriber.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActionPack
+      # ActiveSupport::Notification handlers for ActionController published events
+      # https://guides.rubyonrails.org/active_support_instrumentation.html#action-controller
+      class ActionControllerSubscriber < ActiveSupport::Subscriber
+        def process_action(event)
+          current_span = OpenTelemetry::Trace.current_span
+          return unless current_span.recording?
+
+          attributes = {}
+          add_common_attributes(attributes, event)
+          add_process_action_attributes(attributes, event.payload)
+
+          current_span.add_attributes(attributes) unless attributes.empty?
+        end
+
+        protected
+
+        def add_common_attributes(attributes, event)
+          attach('action_controller.allocations', event.allocations, attributes)
+          attach('action_controller.cpu_runtime_ms', event.cpu_time, attributes)
```

Comment:
**Attribute Naming**
Note how some of the span attributes aren't exactly the same as their event attributes.

I always consider appending units a big help to understanding attributes especially for those seeing them for the first time. I don't know if there is some convention for this though in OTEL world or this repo.